sdk/storage/Azure.Storage.Blobs/tests/BlobClientTests.cs

```diff
@@ -541,6 +538,35 @@ async Task Verify(Task<Response<BlobContentInfo>> upload)
             }
         }
 
+        private static async Task DownloadAndAssert(long size, Stream stream, BlobClient blob)
+        {
+            var actual = new byte[Constants.DefaultBufferSize];
+            using var actualStream = new MemoryStream(actual);
+
+            // reset the stream before validating
+            stream.Position = 0;
+
+            // we are testing Upload, not download: so we download in partitions to avoid the default timeout
+            for (var i = 0; i < size; i += Constants.DefaultBufferSize * 5 / 2)
+            {
+                var startIndex = i;
+                var count = Math.Min(Constants.DefaultBufferSize, (int)(size - startIndex));
+
+                Response<BlobDownloadInfo> download = await blob.DownloadAsync(new HttpRange(startIndex, count));
+                actualStream.Position = 0;
+                await download.Value.Content.CopyToAsync(actualStream);
+
+                var buffer = new byte[count];
+                stream.Position = i;
+                await stream.ReadAsync(buffer, 0, count);
+
+                TestHelper.AssertSequenceEqual(
+                    buffer.AsSpan(0, count).ToArray(),
```

Comment:
Why are we doing the `AsSpan`?  That looks like something that might have been copy/pasted but isn't needed here.